docs/guides/pagination.mdx

```diff
@@ -0,0 +1,234 @@
+
+- **Token-based pagination**: The primary and recommended method for paginating through data in Ory's API.
+- **Offset pagination**: A deprecated method that is still supported but not recommended for new implementations.
+
+In the following sections, we'll explore each of these pagination strategies in detail, with a focus on the recommended
+token-based approach.
+
+## Token pagination
+
+Ory implements token-based pagination as the primary method for handling large datasets in list operations. This approach offers
+better performance and consistency compared to traditional offset-based pagination.
+
+![Token pagination illustration](./_static/pagination/token-pagination.png)
+
+### Default behavior and parameters
+
+By default, Ory paginates list data with a page size of 250 items per page. However, you can customize this behavior using two key
+parameters:
+
+1. `page_size`: Determines the number of items returned per page.
+2. `page_token`: Acts as a pointer to a specific page in the dataset.
+
+### How it works
+
+![Token pointer illustration](./_static/pagination/token-pointer.png)
+
+Let's walk through an example to illustrate how token pagination works in Ory:
+
+Imagine you have 300 customers in your Ory project, and you want to list them 100 at a time.
+
+1. **Initial request**: Set the `page_size` query parameter to 100 for your first request:
+
+   ```shell
+   GET https://$PROJECT_SLUG.projects.oryapis.com/admin/identities?page_size=100
+   Authorization: Bearer $ACCESS_TOKEN
+   ```
+
+2. **Response**: You'll receive a response payload containing the first 100 customers. Additionally, the response header will
+   include a `link` header with details about the first and next pages:
+
+   ```shell
+   link:
+     </admin/identities?page_size=100&page_token=00000000-0000-0000-0000-000000000000>; rel="first",
+     </admin/identities?page_size=100&page_token=30f8507f-40e6-44b9-924f-5f814e3f072e>; rel="next"
+   ```
+
+3. **Understanding the Link Header**:
+
+   - The `first` link always points to the first page of results. Its token (in this case, `00000000-0000-0000-0000-000000000000`)
+     remains constant.
+   - The `next` link points to the next page of results. Its token (e.g., `30f8507f-40e6-44b9-924f-5f814e3f072e`) is unique and
+     changes with each page.
+
+4. **Subsequent requests**: To retrieve the next page, use the `next` token in your follow-up request:
+
+   ```shell
+   GET https://$PROJECT_SLUG.projects.oryapis.com/admin/identities?page_size=100&page_token=30f8507f-40e6-44b9-924f-5f814e3f072e
+   Authorization: Bearer $ACCESS_TOKEN
+   ```
+
+### Best practices
+
+- Always use the tokens provided in the `link` header for navigation. Do not attempt to generate or modify these tokens yourself.
+- If you need to start from the beginning, use the `first` link or omit the `page_token` parameter.
+- Keep track of the previous/current `next` token to allow for backward pagination through your results.
```

Comment:
Add that the user should expect that the max page size can change, and that the page token format will also change and the users should not try to reverse engineer it.